### ISSUE TITLE
"SSH cert not available" on first 'kube credentials' command from a root to leaf cluster

### ISSUE THREAD

**Issue by corkrean** [gravitational/teleport#8090]
### Description

`kube credentials` fails on the first attempt to get credentials from a kube cluster connected to a leaf cluster when command is run from the root cluster. If the command is run again without logging out of Teleport, it works as expected. 

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Run `/usr/local/bin/tsh logout && /usr/local/bin/tsh kube credentials --kube-cluster $leaf_cluster_kube_name --teleport-cluster $leaf_cluster_name --proxy $root_cluster_proxy_url` and notice that it fails. 
2. Run `/usr/local/bin/tsh kube credentials --kube-cluster $leaf_cluster_kube_name --teleport-cluster $leaf_cluster_name --proxy $root_cluster_proxy_url` after the above command and notice that it works fine. 

### Server Details
- Teleport version (run `teleport version`): 7.1.0 (also reported on 6.2.8)
- Server OS (e.g. from `/etc/os-release`): GKE

### Client Details
- Tsh version (`tsh version`): 7.1
- Computer OS (e.g. Linux, macOS, Windows): macOS

### Debug Logs
tsh client logs for first/failed attempt:
```
ERROR REPORT:
Original Error: *trace.NotFoundError SSH cert not available
Stack Trace:
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/lib/client/interfaces.go:367 github.com/gravitational/teleport/lib/client.(*Key).SSHCert
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/lib/client/interfaces.go:279 github.com/gravitational/teleport/lib/client.(*Key).AsAgentKeys
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/lib/client/keyagent.go:175 github.com/gravitational/teleport/lib/client.(*LocalKeyAgent).LoadKey
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/lib/client/keyagent.go:449 github.com/gravitational/teleport/lib/client.(*LocalKeyAgent).AddKey
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/tool/tsh/kube.go:114 main.(*kubeCredentialsCommand).run
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/tool/tsh/tsh.go:607 main.Run
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/tool/tsh/tsh.go:261 main.main
	/usr/local/go/src/runtime/proc.go:225 runtime.main
	/usr/local/go/src/runtime/asm_amd64.s:1371 runtime.goexit
User Message: SSH cert not available
```

debug auth/proxy logs for first/failed attempt:
```
2021-08-30T14:46:44Z DEBU [RBAC]      Grant access to cluster ten.teleport-cs.com - no role in [admin] uses cluster labels and the cluster is not labeled. services/role.go:1365
2021-08-30T14:46:44Z DEBU [KEYGEN]    generated user key for [root] with expiry on (1630377970) 2021-08-31 02:46:10.062516206 +0000 UTC native/native.go:256
2021-08-30T14:46:44Z INFO [CA]        Generating TLS certificate {0x556b168 0xc000e2b070 1.3.9999.1.7=#1317736576656e74682e74656c65706f72742d63732e636f6d,1.3.9999.1.3=#131374656e2e74656c65706f72742d63732e636f6d,CN=test,OU=usage:kube,O=admin,POSTALCODE={\"kubernetes_groups\":[\"\"]\,\"kubernetes_users\":[\"\"]\,\"logins\":null},STREET=ten.teleport-cs.com,L=root 2021-08-31 02:46:10.068770632 +0000 UTC []}. common_name:test dns_names:[] locality:[root] not_after:2021-08-31 02:46:10.068770632 +0000 UTC org:[admin] org_unit:[usage:kube] tlsca/ca.go:637
```

No auth/proxy logs are produced for the second/successful attempt

**Comment by corkrean** [comment]
The same behavior is observed with version 6.0.3 @russjones

**Comment by greedy52** [comment]
Duplicate of #23250
Fixed by #23252
